webapp/src/serialindicator.tsx

```diff
@@ -12,36 +12,43 @@ export interface SerialIndicatorProps {
 
 export interface SerialIndicatorState {
     active?: boolean
+    receivingData?: boolean
 }
 
 export class SerialIndicator extends React.Component<SerialIndicatorProps, SerialIndicatorState>{
+    animationStopper: number
 
     constructor(props: any) {
         super(props)
-        this.state = { active: false }
-        window.addEventListener("message", this.setActive.bind(this))
+        this.state = { active: false, receivingData: false }
+        window.addEventListener("message", this.handleMessage.bind(this))
     }
 
-    setActive(ev: MessageEvent) {
+    handleMessage(ev: MessageEvent) {
         let msg = ev.data
-        if (!this.state.active && msg.type === "serial") {
+        if (msg.type === "serial") {
             const sim = !!msg.sim
             if (sim === this.props.isSim) {
-                this.setState({ active: true })
+                if (!this.state.active) {
+                    this.setState({ active: true, receivingData: true })
+                } else {
+                    clearTimeout(this.animationStopper)
+                }
+                this.animationStopper = setTimeout(() => {this.setState({receivingData: false})}, 1000)
             }
         }
     }
 
     clear() {
-        this.setState({ active: false })
+        this.setState({ active: false, receivingData: false})
     }
 
     render() {
         if (!this.state.active) return <div />;
         return <div className="ui segment inverted serialindicator" tabIndex={0} onClick={this.props.onClick} onKeyDown={sui.fireClickOnEnter}>
             <div className="ui label circular">
                 <div className="detail indicator">
-                    <span className="ui green empty circular label" />
+                    <span className={` ${this.state.receivingData ? "blink" : ""} ui green empty circular label`} />
```

Comment:
nit: extra space before the ${this.state}

Reply:
Fixed :)